Week-23/todo-ui/src/pages/AddTodo.jsx

```diff
@@ -0,0 +1,97 @@
+import React, { useEffect, useState } from "react";
+import styles from "./AddTodo.module.less";
+import axios from "axios";
+
+const AddTodo = () => {
+  const [todo, setTodo] = useState("");
+  const [allTodos, setAllTodos] = useState([]);
+  const [todoToUpdate, setTodoToUpdate] = useState(null);
+
+  const getTodo = async () => {
+    try {
+      const data = await axios.get(`http://localhost:3000/api`);
+      console.log("data", data.data);
+      setAllTodos(data.data);
+    } catch (error) {
+      console.error("Error getting todos:", error);
+    }
+  };
+
+  const handleAddTodo = async () => {
+    try {
+      if (todoToUpdate) {
+        // If todoToUpdate is set, update the existing todo
+        await axios.put(`http://localhost:3000/api/${todoToUpdate._id}`, {
+          todo,
+        });
+        setTodoToUpdate(null); // Reset the todoToUpdate state
+      } else {
+        // Otherwise, add a new todo
+        await axios.post("http://localhost:3000/api", {
+          todo,
+        });
+      }
+      console.log("Todo added/updated successfully!");
+      getTodo();
+      setTodo("");
+    } catch (error) {
+      console.error("Error adding/updating todo:", error);
+    }
+  };
+
+  useEffect(() => {
+    getTodo();
+  }, []);
+
+  const handleDelete = async (id) => {
+    console.log("id", id);
+    if (id) {
+      const updatedTodos = allTodos.find((todo) => todo._id !== id);
+      setAllTodos(updatedTodos);
```

Comment:
This happens even before we call delete api, shouldn’t it only happen if deletion is successful? If it fails, we should show some error that deletion failed on UI.